src/rust/catmem/futures/push.rs

```diff
@@ -73,8 +73,9 @@ impl Future for PushFuture {
     fn poll(self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<Self::Output> {
         let self_: &mut PushFuture = self.get_mut();
         let mut index: usize = self_.index;
-        for x in &self_.buf[index..] {
-            match self_.ring.try_enqueue(*x) {
+        for low in &self_.buf[index..] {
+            let x: u16 = ((0 & 0xff) << 8) | (low & 0xff) as u16;
```

Comment:
Why bother with the `((0 & 0xff) << 8) |` part?  Or really anything other than low as u16?  Not that this will hurt anything though.